packages/hooks/use-archive-infinite-scroll/__tests__/use-archive-infinite-scroll.test.tsx

```diff
@@ -0,0 +1,1098 @@
+      },
+    } as any);
+
+    sourceGet.mockReturnValue({
+      link: "/",
+      isReady: false,
+      isFetching: false,
+    });
+
+    act(() => {
+      render(<App options={{ active: false }} />, container);
+    });
+
+    expect(spiedUseArchiveInfiniteScroll).toHaveBeenCalledTimes(1);
+    expect(routerUpdateState).not.toHaveBeenCalled();
+  });
+
+  test("should return the right object", () => {
+    mockedUseConnect.mockReturnValue({
+      state: {
+        router: {
+          link: "/page/2/",
+          state: {
+            infiniteScroll: {
+              links: ["/", "/page/2/", "/page/3/"],
+            },
+          },
+        },
+        source: {
+          get: sourceGet,
+        },
+      },
+      actions: {
+        router: { updateState: routerUpdateState },
+      },
+    } as any);
+
+    sourceGet.mockReturnValue({
+      link: "/page/3/",
+      isReady: true,
+      isFetching: false,
+    });
+
+    act(() => {
+      render(<App />, container);
+    });
+
+    expect(spiedUseArchiveInfiniteScroll).toHaveBeenCalledTimes(1);
+    expect(spiedUseArchiveInfiniteScroll).toHaveReturnedWith({
+      pages: [
+        {
+          key: "/",
+          link: "/",
+          isLast: false,
+          Wrapper: expect.any(Function),
+        },
+        {
+          key: "/page/2/",
+          link: "/page/2/",
+          isLast: false,
+          Wrapper: expect.any(Function),
+        },
+        {
+          key: "/page/3/",
+          link: "/page/3/",
+          isLast: true,
+          Wrapper: expect.any(Function),
+        },
+      ],
+      isLimit: false,
+      isFetching: false,
+      isError: false,
+      fetchNext: expect.any(Function),
+    });
+  });
+
+  test("should return `isLast` false when the last page is not ready", () => {
+    mockedUseConnect.mockReturnValue({
+      state: {
+        router: {
+          link: "/page/2/",
+          state: {
+            infiniteScroll: {
+              links: ["/", "/page/2/", "/page/3/"],
+            },
+          },
+        },
+        source: {
+          get: sourceGet,
+        },
+      },
+      actions: {
+        router: { updateState: routerUpdateState },
+      },
+    } as any);
+
+    sourceGet.mockReturnValue({
+      link: "/page/3/",
+      isReady: false,
+      isFetching: false,
+    });
+
+    act(() => {
+      render(<App />, container);
+    });
+
+    expect(spiedUseArchiveInfiniteScroll).toHaveBeenCalledTimes(1);
+    expect(spiedUseArchiveInfiniteScroll).toHaveReturnedWith({
+      pages: [
+        {
+          key: "/",
+          link: "/",
+          isLast: false,
+          Wrapper: expect.any(Function),
+        },
+        {
+          key: "/page/2/",
+          link: "/page/2/",
+          isLast: true,
+          Wrapper: expect.any(Function),
+        },
+        {
+          key: "/page/3/",
+          link: "/page/3/",
+          isLast: false,
+          Wrapper: expect.any(Function),
+        },
```

Comment:
I just leaved that code as it is because there is no much of a difference between both approaches. In this case, it wouldn't be probable to have redirected pages – the developer is the one choosing what archive is used as the "source" of posts, and I don't think someone would redirect the `/page/3/` of an archive, for example.

For that reason, as you woudn't have to remove pages from the `pages` array, I didn't change the original implementation.

Reply:
I see. Thanks David. Maybe it is something we can change in future refactorings.